src/pp.rs

```diff
@@ -778,9 +778,7 @@ impl MacroProcessor {
         loop {
             // Get the next token (without additional expansion)
             let token = match self.step_no_continue(lexer) {
-                Err(StepExit::Continue) => {
-                    continue;
-                }
+                Err(StepExit::Continue) => unreachable!(),
```

Comment:
why can't we keep the continue here even if it's unreachable now it might help in the future if we accidentally break this assumption

Reply:
Changed to use `step` instead of `step_no_continue` and put the `continue` right here instead of the `unreachable`.

Reply:
Oh didn't notice the `step_no_continue` ignore this then